targets/TARGET_NUVOTON/TARGET_M2351/analogout_api.c

```diff
@@ -0,0 +1,198 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2015-2016 Nuvoton
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "analogout_api.h"
+
+#if DEVICE_ANALOGOUT
+
+#include "cmsis.h"
+#include "pinmap.h"
+#include "PeripheralPins.h"
+#include "nu_modutil.h"
+
+/* Maximum DAC modules */
+#define NU_DACMOD_MAXNUM        2
+/* Maximum DAC channels per module */
+#define NU_DACCHN_MAXNUM        1
+
+static uint32_t dac_modinit_mask[NU_DACMOD_MAXNUM];
+
+static const struct nu_modinit_s dac_modinit_tab[] = {
+    {DAC_0_0, DAC_MODULE, 0, 0, DAC_RST, DAC_IRQn, NULL},
+    {DAC_1_0, DAC_MODULE, 0, 0, DAC_RST, DAC_IRQn, NULL}
+};
+
+void analogout_init(dac_t *obj, PinName pin)
+{   
+    obj->dac = (DACName) pinmap_peripheral(pin, PinMap_DAC);
+    MBED_ASSERT(obj->dac != (DACName) NC);
+
+    const struct nu_modinit_s *modinit = get_modinit(obj->dac, dac_modinit_tab);
+    MBED_ASSERT(modinit != NULL);
+    MBED_ASSERT(modinit->modname == obj->dac);
```

Comment:
`get_modinit(...)` can't return `modinit` object with` modname != obj->dac`.
So, `MBED_ASSERT(modinit->modname == obj->dac)` is redundancy.

Reply:
@cyliangtw This is to check the flow `PinMap_DAC`/`dac_modinit_tab`/`get_modinit` is bug-less, like error in any of them, in development time. Anyway, `MBED_ASSERT` is removed in release build.